### ISSUE TITLE
Throw new TypeError during npm install

### ISSUE THREAD

**Issue by mju516** [imagemin/gifsicle-bin#61]
First, I would like to apologize in advance. I am a complete novice when it comes to node, and I've basically been following "step-by-step guides" up until this point. I'm attempting to run "npm install", and I haven't been able to resolve this issue on my own.

During the run, I get the following error:

<hr>
node index.js

path.js:7
throw new TypeError('Path must be a string. Received ' + inspect(path)); Received {url: 'https://raw.github.com/imagemin/gifsicle-bin/v0.1.7/vendor/win/x8/gifsicle.exe', 
name: 'gifsicle.exe',
os: 'win32',
arch: 'x64'} 
at assertPath (path.js:7:11)
at Object.basename (path.js:801:5)
<hr>

And then:

<hr>
gifsicle@0.1.7 postinstall: 'node index.js'
Exit Status 1

Failed at the gifsicle@0.1.7 postinstall script 'node index.js'
<hr>

And the npm instructions on how to get here. I've installed a newer version of gifsicle and update the pacakge.json (3.0.3), but that doesn't seem to have done anything. Is there some simple configuration change that I may have forgotten to make? I've included the relevant(?) output from the generated npm-debug.log file.

<hr>
Error: gifsicle@0.1.7 postinstall: `node index.js`
Exit status 1
at EventEmitter.<anonymous> (C:\Users\matt\AppData\Roaming\npm\node_modules\npm\lib\utils\lifecycle.js:242:16)
at emitTwo (events.js:106:13)
at EventEmitter.emit (events.js:191:7)
at ChildProcess.<anonymous> (C:\Users\matt\AppData\Roaming\npm\node_modules\npm\lib\utils\spawn.js:40:14)
at emitTwo (events.js:106:13)
at ChildProcess.emit (events.js:191:7)
at maybeClose (internal/child_process.js:852:16)
at Process.ChildProcess._handle.onexit (internal/child_process.js:215:5)
pkgid gifsicle@0.1.7
cwd c:\users\matt\documents\visual studio 2015\Projects\Avenger_APP_test\Avenger_APP_test
Windows_NT 10.0.10586
argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\matt\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install"
node v6.2.1
npm  v3.10.2
code ELIFECYCLE
gifsicle@0.1.7 postinstall: `node index.js`
Exit status 1
Failed at the gifsicle@0.1.7 postinstall script 'node index.js'.
Make sure you have the latest version of node.js and npm installed.
If you do, this is most likely a problem with the gifsicle package,
not with npm itself.
Tell the author that this fails on your system:
    node index.js

**Comment by shinnn** [comment]
> @0.1.7

Ancient version. Upgrade gifsicle-bin.

**Comment by mju516** [comment]
@shinnn Per my post, I tried that. 

"I've installed a newer version of gifsicle and update the pacakge.json (3.0.3)"

To confirm, running "npm ls gifsicle" returns "3.0.3". Hence my confusion as to why it's yelling at me for having 0.1.17.

**Comment by shinnn** [comment]
The generated npm-debug.log file tells us you're still trying to install gifsicle@0.1.7.

**Comment by mju516** [comment]
Right, but I cannot understand why. So I checked the cache, and both the 3.0.3 and the 0.1.7 files were in there. So I cleared them, and reran "npm install". Only the gifsicle@0.1.7 version was back, even though I have gifsicle@3.0.3 in my package.json file. 

I've searched through my entire project, there are no other references to gifsicle. I do not understand why it is still installing the 0.1.7. Is there some other file outside of my project that I'm not aware of that could be causing this? Or some obvious setting that I may be missing?
